src/main/java/de/tum/in/www1/artemis/repository/TextBlockRepository.java

```diff
@@ -30,4 +32,17 @@
     Set<TextBlock> findAllBySubmissionIdIn(Set<Long> submissionIdList);
 
     void deleteAllBySubmission_Id(Long submissionId);
+
+    /**
+     * For the given TextBlock `id` finds it's respective cluster and retrieves the number of other blocks
+     * @param id the id of the TextBlock
+     * @return the number of other blocks in the same cluster as the block with given `id`
+     */
+    @Query("""
+            SELECT COUNT(c) - 1
+            FROM TextBlock c
+            WHERE c.cluster.id IN
+            (SELECT DISTINCT cluster.id FROM TextBlock WHERE id = :#{#id} )""")
```

Comment:
Does this need a nested query?
from some experiments, this seems to be quicker (8ms)
```sql
select count(distinct tb.id) from text_block b
left join text_cluster tc on b.cluster_id = tc.id
left join text_block tb on tc.id = tb.cluster_id and tb.id != b.id
where b.id = "a674f0ce1dc446080a0afbbbf8df6221741b7782"
```
compared to this (10ms)
```sql
SELECT (COUNT(c.id) - 1) as countval
FROM text_block c
WHERE c.cluster_id IN
      (SELECT DISTINCT cluster_id FROM text_block WHERE id = "a674f0ce1dc446080a0afbbbf8df6221741b7782" )
```

Reply:
Added the suggested change 👍